### PR TITLE
[android_intent] add flags parameter

### PR DESCRIPTION
**Pull request by datvo0110**
## Description
we are launching to a different package we need to set the FLAG_ACTIVITY_NEW_TASK flag

Similar to react native Link.openUrl() : https://github.com/facebook/react-native/blob/3b6f6ca4d5fcee6f1bc6d6242e3e2ef136e4d546/ReactAndroid/src/main/java/com/facebook/react/modules/intent/IntentModule.java#L97

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by datvo0110** [comment]
> @googlebot I signed it!

**Comment by ened** [comment]
@datvo0110 please bump version & add change log entry.

I'm wondering if this should be done explicitly via an extra enum flag in the AndroidIntent class.
Can you think of other flags that may be of interest in the future?

I can think of those, for example:

```
Intent.FLAG_ACTIVITY_NEW_TASK
Intent.FLAG_ACTIVITY_NO_HISTORY
Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
Intent.FLAG_GRANT_READ_URI_PERMISSION
```

**Comment by datvo0110** [comment]
@ened I have bumped version from 0.3.2 to 0.3.3 & add change log entry. Is this ok?

**Comment by ened** [comment]
@datvo let’s rewrite this a bit and add the flags argument instead of having app behavior done implicitly using the package name only. L

Are you up for changing adding and implementing the ‘flags’ parameter?

**Comment by datvo0110** [comment]
ok, let's me do that! I'll try to cover most case :D

**Comment by datvo0110** [comment]
I added flags parameter as List<String> and prepare some popular flags as
you recommended

Intent.FLAG_ACTIVITY_NEW_TASK
Intent.FLAG_ACTIVITY_NO_HISTORY
Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
Intent.FLAG_GRANT_READ_URI_PERMISSION

I also added an example on how to use flags option.
I'm looking for your opinions :D

File: packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java [hunk 1]
```diff
@@ -51,6 +51,29 @@ private String convertAction(String action) {
     }
   }
 
+  private int convertFlag(String flag) {
```
Comment (ened) [inline packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java]: This PR converts the flags from String to integer using this method. 
A better method is to let the Dart side know about the flag names & values and simply push the flag through. You can create a new Flags.dart (or similar) and fill in the constants from https://developer.android.com/reference/android/content/Intent (All the constants from FLAG_).

Please follow the hex numbering (0x00400000) and keep the order as it is listed on the documentation page.

File: packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java [hunk 2]
```diff
@@ -51,6 +51,29 @@ private String convertAction(String action) {
...
+      case "exclude_from_recents":
+        return Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS;
+      case "grant_read_uri_permission":
+        return Intent.FLAG_GRANT_READ_URI_PERMISSION;
+      default:
+        return 0;
+    }
+  }
+
+  private int convertFlags(ArrayList<String> flags) {
```
Comment (ened) [inline packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java]: As said above, this logic should be done in the Dart side.

File: packages/android_intent/example/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -9,7 +9,6 @@
 /* Begin PBXBuildFile section */
 		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
 		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
-		2D5378261FAA1A9400D5DBA9 /* flutter_assets in Resources */ = {isa = PBXBuildFile; fileRef = 2D5378251FAA1A9400D5DBA9 /* flutter_assets */; };
```
Comment (ened) [inline packages/android_intent/example/ios/Runner.xcodeproj/project.pbxproj]: The changes in here relate to the iOS example. If possible, please remove them from the commit.

File: packages/android_intent/example/lib/main.dart [hunk 4]
```diff
@@ -117,6 +117,16 @@ class ExplicitIntentsWidget extends StatelessWidget {
     intent.launch();
   }
 
+  void _openLinkInDefaultBrowserInSeperatedWindow() {
+    final List<String> flags = <String>[];
+    flags.add('new_task');
+    final AndroidIntent intent = AndroidIntent(
+        action: 'action_view',
+        data: Uri.encodeFull('https://flutter.io'),
+        flags: flags);
```
Comment (ened) [inline packages/android_intent/example/lib/main.dart]: - please add a `,` to before the parentheses end
- this now becomes simpler as you can do the bit operation in Dart

File: packages/android_intent/example/lib/main.dart [hunk 5]
```diff
@@ -162,6 +172,11 @@ class ExplicitIntentsWidget extends StatelessWidget {
                 child: const Text('Tap here to open link in Google Chrome.'),
                 onPressed: _openLinkInGoogleChrome,
               ),
+              RaisedButton(
+                child: const Text(
+                    'Tap here to open link in browser in a seperated window.'),
+                onPressed: _openLinkInDefaultBrowserInSeperatedWindow,
```
Comment (ened) [inline packages/android_intent/example/lib/main.dart]: I like a close-to-reality example like this one, however I would name it more accurate ("Start Activity in New Task"). What other ways could we use in the example to demonstrate the functionality?

**Comment by datvo0110** [comment]
@ened I already
-- bring logic from java to dart, add flag list
-- add test case
-- update example
sorry because I tried to remove packages/android_intent/example/ios/Runner.xcodeproj/project.pbxproj so I have to reset to the old commit, now it contained some unrelated file @@

**Comment by datvo0110** [comment]
@ened 
sorry, the commit is like a mess now
i have already prepared a new clean commit https://github.com/flutter/plugins/pull/2000
let's switch to the other commit to continue

**Comment by ened** [comment]
@datvo0110 ok let's close this one and move to #2000.

**Comment by datvo0110** [comment]
Hi @ened, @mklim please help me review https://github.com/flutter/plugins/pull/2000

**Pull request closed without merging**